### PR TITLE
Email service and reset password

### PR DESCRIPTION
**Pull request by Mattia-Marino**
New adds and changes:
- Added email service based on Gmail API. Useful to send notifications to the user, registration validation, reset password and more.
- Added reset password feature. If a registered user forgot the password to access to the platform can click on "Forgot password" and be brought to a page where the user can insert the email used to register. An email will then be sent to said email containing a link that can be used for password reset. The link will be valid for 15 minutes.
- More minor fixes

### CODE REVIEW

File: qualitag/src/main/java/it/unisannio/studenti/qualitag/security/filter/JwtAuthenticationFilter.java [hunk 1]
```diff
@@ -45,7 +46,15 @@ protected void doFilterInternal(HttpServletRequest request,
 
     jwt = authHeader.substring(7);
     log.debug("JWT - {}", jwt);
-    userIdentifier = jwtService.extractUserName(jwt);
+
+    try {
+      userIdentifier = jwtService.extractUserName(jwt);
+    } catch (ExpiredJwtException e) {
```
Comment (follen99) [inline qualitag/src/main/java/it/unisannio/studenti/qualitag/security/filter/JwtAuthenticationFilter.java:52]: Nice addition!

Comment (Mattia-Marino) [reply qualitag/src/main/java/it/unisannio/studenti/qualitag/security/filter/JwtAuthenticationFilter.java:52]: Thanks mate

File: qualitag/src/main/resources/application.properties [hunk 2]
```diff
@@ -11,4 +11,7 @@ spring.data.mongodb.uri=mongodb://localhost:27017/qualitag
 token.secret.key=48a868a4042f634ac04a117f00a87202131dd7c46c4b32c4acb3edc5e15f4511
 
 # JWT expiration is 1 hour
-token.expiration=3600000
\ No newline at end of file
+token.expiration=3600000
+
+# JWT password reset token is 15 minutes
+token.password.expiration=15
```
Comment (follen99) [inline qualitag/src/main/resources/application.properties:17]: Interesting! maybe we can communicate the token lifespan to the user via the mail, such as: "The link will last 15 minutes". (Nice to have addition)

**Review by follen99 — APPROVED** [review]
## Everything looks ok!
:kiss: :eggplant:

**Pull request merged**